src/auth/auth.ts

```diff
@@ -36,6 +38,22 @@ export class BBAuth {
     }
   } = {};
 
+  public static getUrl(
```

Comment:
Should this method's name be something more specific to indicate that it generates a "zoned" URL?  Like `getZonedUrl()`?

Reply:
I can go either way.  I think generally (for now at least) all URLs are zoned, and this method is really providing 2 functions:

- Convert generic 1bb scheme address to real addresses
- Inject the zone portion in

If we go to support non-zoned urls by some means, I'd imagine it would somehow be articulated in the 1bb schemed url that is passed into $http rather than using a different function?  Thoughts?

Reply:
Now I see the zone being passed in might make it weird to use w/o a zone, haha.  Hrmmmmm.

I could factor the args out into a GetUrlArgs so its less directly oriented around zones.

Reply:
Yeah, I like the idea of an `args` parameter.  I'd probably keep `url` as its own parameter since it's required, and put optional parameters on `args`.  If we're sticking with `getUrl()`, maybe we should name the `url` parameter something more descriptive, like `tokenizedUri` to indicate it's not a standard web address.  Having said that, we should probably also rename other variables to use `uri` rather than `url`.